File: readability.c
// Assessing the readibility (in grades) of text using Coleman-Liau formula
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
// Get user input
    string user_text = get_string("Please type in the text you want assessed: ");
// Count the number of letters
    int letters = 0;
    for (int i = 0, n = strlen(user_text); i < n; i++)
    {
        if ((user_text[i] >= 'a' && user_text[i] <= 'z') || (user_text[i] >= 'A' && user_text[i] <= 'Z'))
        {
            letters++;
        }
        else {}
    }

// Count the number of sentences
    int sentences = 0;
    for (int i = 0, n = strlen(user_text); i < n; i++)
    {
        if (user_text[i] == '.' || user_text[i] == '!' || user_text[i] == '?')
        {
            sentences++;
        }
        else {}
    }

// Count the number of words
    int words = 0;
    for (int i = 0, n = strlen(user_text); i < n; i++)
    {
        if (user_text[i] == ' ')
        {
            words++;
        }
        else {}
    }
    words++;

// Calculate the Coleman-Liau index
    float CL_index = (0.0588 * 100 * letters / words) - (0.296 * 100 * sentences / words) - 15.8;
    if (CL_index >= 1 && CL_index <= 16)
    {
        printf("Grade %.0f\n", CL_index);
    }
    else if (CL_index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}

File: readability.py
from cs50 import get_string

# Get user input
user_text = get_string("Please type in the text you want assessed: ")

# Count the number of letters, sentences and words
letters = 0
sentences = 0
words = 1

for i in user_text:
    if i.isalpha():
        letters += 1
    elif(i == "." or i == "!" or i == "?"):
        sentences += 1
    elif i == " ":
        words += 1
    
# Calculate the Coleman-Liau index
CL_index = (0.0588 * 100 * letters / words) - (0.296 * 100 * sentences / words) - 15.8

if CL_index >= 1 and CL_index <=16:
    print(f"Grade {round(CL_index)}")
elif CL_index <1:
    print("Before Grade 1")
else:
    print("Grade 16+")

